src/elfpy/pricing_models/base.py

```diff
@@ -438,13 +435,99 @@ def get_max_long(
                 ):
                     bond_percent -= step_size
                 else:
+                    last_maybe_max_long = (maybe_max_long, d_bonds)
                     if bond_percent == 1:
-                        return maybe_max_long
-                    last_maybe_max_long = maybe_max_long
+                        return last_maybe_max_long
                     bond_percent += step_size
 
         return last_maybe_max_long
 
+    def get_max_short(
```

Comment:
shouldn't the pricing model get_max_short be exact, if you assume an agent unconstrained by base in their wallet?

Reply:
is it because this allows us to check your invariants? if we can verify those are not needed, then we could switch to the closed form solution, correct? although that's not much of an improvement, since you use it as your starting point here anyway. what you have here is simply more general and covers more ground. no need to move away from it.